### PR TITLE
Logging subsystem

### PR DESCRIPTION
**Pull request by fnoble**
WIP

### CODE REVIEW

File: include/libswiftnav/logging.h [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ *    #define DEBUG 1
+ *
+ * \{ */
+
+/** Log an error.
+ * \param args `printf` style format and arguments.
+ */
+#define log_error(args...) printf("ERROR: " args)
+
```
Comment (denniszollo) [inline include/libswiftnav/logging.h]: I think perhaps we should make these macro functions all caps, as to help indicate to the code reader that they are macros. The Kernel standards say this is optional, but I like to do it. 
excerpt from kernel coding standards
 "CAPITALIZED macro names are appreciated but macros resembling functions
may be named in lower case."

Comment (fnoble) [inline include/libswiftnav/logging.h]: Usually I would agree, but I think there is a good chance they may not be macros in the final implementation...

File: src/amb_kf.c [hunk 2]
```diff
@@ -22,14 +22,16 @@
...
+#include "logging.h"
+#include "linear_algebra.h"
 #include "constants.h"
 #include "track.h"
 #include "almanac.h"
 #include "gpstime.h"
 #include "amb_kf.h"
 
-#define DEBUG_AMB_KF 0
+#define DEBUG 0
```
Comment (denniszollo) [inline src/amb_kf.c]: Don't we want to define DEBUG before including logging.h? If I change this debug to 1, I think the compiler will warm me about multiple definitions of the macro.

**Comment by gsmcmullin** [comment]
Looks good.  A big improvement.

**Pull request merged**